news.js

```diff
@@ -0,0 +1,250 @@
+		popup:
+			"A throw from deep caught an edge of Stokes' bat and the ball raced away for a four. It's hard to imagine a batsman scoring runs with the bat while sliding down to save himself from getting run-out. But that's what happened in the epic finale. Stunned by what just happened, New Zealand players saw their worst nightmare come true as England got six runs (two by running and four extra). Despite just three runs needed off the final two balls, the Kiwis were as spirited as ever and thanks to their nerves, England could only manage to get two of the remaining two balls to end the contest in a tie."
+	},
+	{
+		image: "<div class='content__img'></div>",
+		heading: 'Fox News',
+		date: 'Posted on 29 June, 2019 //Category: Category One',
+		content: 'Djokovic reveals ‘silly’ mental trick to help him deal with crowd at Wimbledon',
+		popup: `Djokovic revealed a mental trick, according to the Times. After winning the match, after nearly five hours, he told reporters that when the crowd chanted “Roger! Roger,” he said he heard “Novak.”“It sounds silly, but it is like that. I try to convince myself that it’s like that.” The sets were 7-6 (5), 1-6, 7-6 (4), 4-6 and 13-12 (3). The final lasted nearly five hours, longer than any other Wimbledon final. Djokovic has now won Wimbledon five times.`
+	},
+	{
+		image: "<div class='content__img'></div>",
+		heading: 'CNN',
+		date: 'Posted on 29 June, 2019 //Category: Category One',
+		content: "'Silverstone, I love you': Lewis Hamilton wins record sixth British Grand Prix",
+		popup: `Lewis Hamilton claimed a thrilling victory at Silverstone on Sunday to win a record-breaking sixth British Grand Prix. It looked as though teammate Valtteri Bottas would spoil the party for the 141,000 visiting fans, but a timely safety car intervention gifted Hamilton first place. Ferrari's Charles Leclerc claimed an impressive third place after an intense, wheel-to-wheel battle with Max Verstappen throughout the race.`
+	}
+];
+//Channels present in the dropdown list
+let channels = ['BBC', 'Aaj Tak', 'NBC', 'NDTV', 'Republic', 'TV9', 'India Today', 'Times Now', 'Fox News', 'CNN'];
+//variable used to store the complete data of all news channels
+let fullData = ' ';
+//variable used to store the popup content
+let modalData = `<div id="myModal" class="modal">
+					<div class="modal-content">
+						<div class="modal-header">
+							<span class="close" onclick="closepopup()">&times;</span>
+							<h2 id="popup_head"></h2>
+						</div>
+						<div class="modal-body">
+							<p id="popup_content"></p>
+						</div>
+						<div class="modal-footer">
+							<h3 id="popup_foot">&copy; NewsFeed 2019</h3>
+						</div>
+					</div>
+				</div>`;
+
+//Function to display the selected news category
+function myFunction() {
+	let selectedCategory = document.getElementById('sel-category').value;
+	let display;
+	let indexDisplay;
+	if (selectedCategory !== 'All') {
+		let singleData = data.filter(function (e, index) {
```

Comment:
in filter return only the data / index.  return selectedCategory === e.heading line is sufficient

Reply:
showpopup takes the index of JSON object as an argument so I'm storing the index in filter to pass it to showpopup function.